### PR TITLE
Feature/48062 Migrate Table

### PR DESCRIPTION
**Pull request by Aekal**

### CODE REVIEW

File: components/02-elements/table/Table.vue [hunk 1]
```diff
@@ -0,0 +1,300 @@
+<template>
+  <table
+    class="table"
+    :class="className"
```
Comment (jszczech) [inline components/02-elements/table/Table.vue]: `:class="['table', className]"` :)

File: components/02-elements/table/Table.vue [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    className: {
+      type: String,
+      default: null
+    },
+    caption: {
+      type: String,
+      default: null
+    },
+    headCells: {
+      type: Array,
```
Comment (jszczech) [inline components/02-elements/table/Table.vue]: I think, headCells should be required.

File: components/02-elements/table/Table.vue [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    role="table"
+  >
+    <caption
+      v-if="caption"
+      class="table__caption"
+    >
+      {{ caption }}
+    </caption>
+
+    <thead v-if="headCells">
```
Comment (jszczech) [inline components/02-elements/table/Table.vue]: I think, heading should be always visible. Maybe if `bodyRows` is missing, we should display some placeholder with information? What do you think?

Comment (Aekal) [reply components/02-elements/table/Table.vue]: I set both thead and tbody to be required, imo we shouldn't render empty table (without content)

**Pull request merged**